--- obconf.orig/src/preview.c	2011-08-05 23:05:10.000000000 +0700
+++ obconf/src/preview.c	2011-12-23 10:51:36.834671883 +0700
@@ -322,55 +322,8 @@
                                                   0, 0, x, y, w, h);
 
             x += w + theme->paddingx + 1;
-        } else {
-            /* buttons */
-            switch (*layout) {
-            case 'D':
-                a = focus ?
-                    theme->btn_desk->a_focused_unpressed :
-                    theme->btn_desk->a_unfocused_unpressed;
-                break;
-            case 'S':
-                a = focus ?
-                    theme->btn_shade->a_focused_unpressed :
-                    theme->btn_shade->a_unfocused_unpressed;
-                break;
-            case 'I':
-                a = focus ?
-                    theme->btn_iconify->a_focused_unpressed :
-                    theme->btn_iconify->a_unfocused_unpressed;
-                break;
-            case 'M':
-                a = focus ?
-                    theme->btn_max->a_focused_unpressed :
-                    theme->btn_max->a_unfocused_unpressed;
-                break;
-            case 'C':
-                a = focus ?
-                    theme->btn_close->a_focused_unpressed :
-                    theme->btn_close->a_unfocused_unpressed;
-                break;
-            default:
-                continue;
-            }
-
-            a->surface.parent = title;
-            a->surface.parentx = x - theme->fbwidth;
-            a->surface.parenty = theme->paddingy + 1;
-
-            w = theme->button_size;
-            h = theme->button_size;
-
-            theme_pixmap_paint(a, w, h);
-            pixmap = gdk_pixmap_foreign_new(a->pixmap);
-            /* use y + 1 because these buttons should be centered wrt the label
-             */
-            pixbuf = gdk_pixbuf_get_from_drawable(pixbuf, pixmap,
-                                                  gdk_colormap_get_system(),
-                                                  0, 0, x, y + 1, w, h);
-
-            x += theme->button_size + theme->paddingx + 1;
         }
+
     }
 
     if (theme->handle_height) {
@@ -512,14 +465,14 @@
         return NULL;
 
     menu = preview_menu(theme);
-  
+
     window_w = theme_window_min_width(theme, titlelayout);
 
     menu_w = gdk_pixbuf_get_width(menu);
     h = gdk_pixbuf_get_height(menu);
 
     w = MAX(window_w, menu_w) + 20;
-  
+
     /* we don't want windows disappearing on us */
     if (!window_w) window_w = menu_w;
 
