--- Mesa-9.2.5/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp.orig	2014-05-14 00:06:21.396551089 +0000
+++ Mesa-9.2.5/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2014-05-14 00:09:26.705974965 +0000
@@ -275,8 +275,10 @@
    TargetOptions options;
 #if defined(PIPE_ARCH_X86)
    options.StackAlignmentOverride = 4;
+#if HAVE_LLVM < 0x0304
    options.RealignStack = true;
 #endif
+#endif
 
 #if defined(DEBUG)
    options.JITEmitDebugInfo = true;
