diff -up gpac/configure.ogl_libs gpac/configure
--- gpac/configure.ogl_libs	2009-09-19 13:29:46.988993098 +0200
+++ gpac/configure	2009-09-19 13:33:34.840741932 +0200
@@ -1319,7 +1319,7 @@ if test "$disable_3d" = "no" ; then
   elif test "$darwin" = "yes" ; then
     LINK3D="-lgl -lglu"
   else
-   LINK3D="-lGL -lglut"
+   LINK3D="-lGLU -lGL -lglut"
   fi
   if $cc -o $TMPO $TMPC $LINK3D 2> /dev/null  ; then
     has_opengl="yes"
