diff -up gpac/configure.lib64 gpac/configure
--- gpac/configure.lib64	2008-12-17 14:25:31.000000000 +0100
+++ gpac/configure	2008-12-17 14:26:24.000000000 +0100
@@ -1126,10 +1126,10 @@ if test "$disable_opengl" = "no" ; then
   fi
   if $cc -o $TMPO $TMPC $LINK3D 2> /dev/null  ; then
     has_opengl="yes"
-  elif $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib 2> /dev/null  ; then
+  elif $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib64 2> /dev/null  ; then
     has_opengl="yes"
     INCL3D="-I$X11_PATH/include"
-    LINK3D="-L$X11_PATH/lib $LINK3D"
+    LINK3D="-L$X11_PATH/lib64 $LINK3D"
   fi
   if test "$has_opengl" = "no" ; then
    LINK3D=""
@@ -1186,7 +1186,7 @@ cat > $TMPC << EOF
 #include <X11/Xlib.h>
 int main( void ) { return 0; }
 EOF
-if $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib 2> /dev/null  ; then
+if $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib64 2> /dev/null  ; then
 has_x11="yes"
 
 #look for X11 shared memory support
@@ -1198,7 +1198,7 @@ cat > $TMPC << EOF
 int main( void ) { return 0; }
 EOF
 
-if $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib 2> /dev/null  ; then
+if $cc -o $TMPO $TMPC -I$X11_PATH/include -L$X11_PATH/lib64 2> /dev/null  ; then
 has_x11_shm="yes"
 fi
 
