diff -up gpac/applications/generators/SVG/Makefile.shared gpac/applications/generators/SVG/Makefile
--- gpac/applications/generators/SVG/Makefile.shared	2007-01-09 13:43:57.000000000 +0100
+++ gpac/applications/generators/SVG/Makefile	2008-02-17 17:56:49.000000000 +0100
@@ -15,7 +15,7 @@ LDFLAGS+=-pg
 endif
 
 #common obj
-OBJS= html.o laser.o main.o v1.o v2.o v3.o ../../../src/utils/list.o ../../../src/utils/error.o
+OBJS= html.o laser.o main.o v1.o v2.o v3.o
 
 CFLAGS+=-g
 LDFLAGS+=-g
@@ -24,10 +24,12 @@ CFLAGS+=$(XML2_CFLAGS)
 ifeq ($(CONFIG_WIN32),yes)
 EXE=.exe
 PROG=SVGGen$(EXE)
+OBJS= html.o laser.o main.o v1.o v2.o v3.o ../../../src/utils/list.o ../../../src/utils/error.o
 EXTRALIBS+=-lwsock32 -lz
 else
 EXT=
 PROG=SVGGen
+LIBS=-L../../../bin/gcc/ -lgpac
 endif
 
 SRCS := $(OBJS:.o=.c) 
@@ -35,7 +37,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 SVGGen$(EXE): $(OBJS) 
-	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(XML2_LFLAGS) $(EXTRALIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(XML2_LFLAGS) $(EXTRALIBS)
 
 
 %.o: %.c
diff -up gpac/applications/generators/X3D/Makefile.shared gpac/applications/generators/X3D/Makefile
--- gpac/applications/generators/X3D/Makefile.shared	2006-05-24 11:52:52.000000000 +0200
+++ gpac/applications/generators/X3D/Makefile	2008-02-17 17:56:09.000000000 +0100
@@ -15,15 +15,16 @@ LDFLAGS+=-pg
 endif
 
 #common obj
-OBJS= main.o ../../../src/utils/list.o  ../../../src/utils/error.o
+OBJS= main.o
 
 ifeq ($(CONFIG_WIN32),yes)
 EXE=.exe
 PROG=X3DGen$(EXE)
+OBJS= main.o ../../../src/utils/list.o  ../../../src/utils/error.o
 else
-OBJS+=../../../src/utils/os_divers.o
 EXT=
 PROG=X3DGen
+LIBS=-L../../../bin/gcc/ -lgpac
 endif
 
 SRCS := $(OBJS:.o=.c) 
@@ -31,7 +32,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 $(PROG): $(OBJS) 
-	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(EXTRALIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(EXTRALIBS)
 
 
 %.o: %.c
diff -up gpac/applications/generators/MPEG4/Makefile.shared gpac/applications/generators/MPEG4/Makefile
--- gpac/applications/generators/MPEG4/Makefile.shared	2006-05-24 11:52:52.000000000 +0200
+++ gpac/applications/generators/MPEG4/Makefile	2008-02-17 17:55:25.000000000 +0100
@@ -15,15 +15,16 @@ LDFLAGS+=-pg
 endif
 
 #common obj
-OBJS= main.o ../../../src/utils/list.o ../../../src/utils/error.o
+OBJS= main.o 
 
 ifeq ($(CONFIG_WIN32),yes)
 EXE=.exe
 PROG=MPEG4Gen$(EXE)
+OBJS= main.o ../../../src/utils/list.o ../../../src/utils/error.o
 else
-OBJS+=../../../src/utils/os_divers.o
 EXT=
 PROG=MPEG4Gen
+LIBS=-L../../../bin/gcc/ -lgpac
 endif
 
 SRCS := $(OBJS:.o=.c) 
@@ -31,7 +32,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 $(PROG): $(OBJS) 
-	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(EXTRALIBS) 
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(EXTRALIBS) 
 
 
 %.o: %.c
