=== modified file 'doc/stellarium.pod.cmake'
--- doc/stellarium.pod.cmake	2012-03-19 15:10:01 +0000
+++ doc/stellarium.pod.cmake	2013-05-27 02:15:10 +0000
@@ -183,9 +183,9 @@
 
 =head1 AUTHOR
 
-Fabien Chéreau, Rob Spearman, Johan Meuris, Matthew Gates, 
+Fabien Chereau, Rob Spearman, Johan Meuris, Matthew Gates, 
 Johannes Gajdosik, Nigel Kerr, Andras Mohari, Bogdan Marinov, 
-Timothy Reaves, Mike Storm, Diego Marcos, Guillaume Chéreau, 
+Timothy Reaves, Mike Storm, Diego Marcos, Guillaume Chereau, 
 Alexander Wolf, Georg Zotti
 
 x14817

