--- empathy-3.4.2.3/libempathy/cheese-camera-device-monitor.c.orig	2012-10-30 16:15:57.050046491 +0700
+++ empathy-3.4.2.3/libempathy/cheese-camera-device-monitor.c	2012-10-30 16:17:21.681060901 +0700
@@ -339,10 +339,9 @@
 {
   GObjectClass *object_class = G_OBJECT_CLASS (klass);
 
-  if (empathy_device_monitor_cat == NULL)
-    GST_DEBUG_CATEGORY_INIT (empathy_device_monitor_cat,
-                             "empathy-device-monitor",
-                             0, "Empathy Camera Device Monitor");
+  GST_DEBUG_CATEGORY_INIT (empathy_device_monitor_cat,
+                           "empathy-device-monitor",
+                           0, "Empathy Camera Device Monitor");
 
   object_class->finalize = empathy_camera_device_monitor_finalize;
 
