diff -up tvtime-1.0.2/src/videoinput.c.videoinput tvtime-1.0.2/src/videoinput.c
--- tvtime-1.0.2/src/videoinput.c.videoinput	2010-11-24 10:31:47.940195589 +0100
+++ tvtime-1.0.2/src/videoinput.c	2010-11-24 10:31:47.968195777 +0100
@@ -377,6 +377,7 @@ uint8_t *videoinput_next_frame( videoinp
         wait_for_frame_v4l2( vidin );
  
         cur_buf.type = vidin->capbuffers[ 0 ].vidbuf.type;
+        cur_buf.memory = vidin->capbuffers[ 0 ].vidbuf.memory;
         if( ioctl( vidin->grab_fd, VIDIOC_DQBUF, &cur_buf ) < 0 ) {
             /* some drivers return EIO when there is no signal */
             if( errno != EIO ) {
