diff --git a/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp b/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
index be265c1..39f14e6 100644
--- a/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
+++ b/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
@@ -144,6 +144,8 @@ uint8_t AUDMEncoder_Lavcodec::initialize(void)
   CONTEXT->channels     =  _wavheader->channels;
   CONTEXT->sample_rate  =  _wavheader->frequency;
   CONTEXT->bit_rate     = (lavConfig.bitrate*1000); // bits -> kbits
+  CONTEXT->sample_fmt=AV_SAMPLE_FMT_S16;
+  CONTEXT->request_sample_fmt=AV_SAMPLE_FMT_S16;
 
   AVCodec *codec;
   CodecID codecID;
