Description: Avoid MAKEFLAGS in Makefile
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2010-03-09

--- a/Makefile
+++ b/Makefile
@@ -85,7 +85,7 @@
 dist-copy-files:
 	cp $(dist_files) $(dist_dir)
 	for i in $(dist_subdirs); do \
-		$(MAKE) dist-copy-files -C $$i $(MAKEFLAGS) \
+		$(MAKE) dist-copy-files -C $$i \
 		  dist_dir=`pwd`/$(dist_dir); \
 	done
 
@@ -102,10 +102,10 @@
 check: test
 
 test:
-	$(MAKE) test -C test $(MAKEFLAGS) PNGPP=`pwd`
+	$(MAKE) test -C test PNGPP=`pwd`
 
 test-clean:
-	$(MAKE) clean -C test $(MAKEFLAGS)
+	$(MAKE) clean -C test
 
 test-compile-headers: *.hpp
 	for i in *.hpp; do \
@@ -135,10 +135,10 @@
 	rm -rf $(PREFIX)/share/doc/$(dist_dir) $(PREFIX)/share/doc/png++
 
 examples:
-	$(MAKE) -C example $(MAKEFLAGS)
+	$(MAKE) -C example
 
 examples-clean:
-	$(MAKE) clean -C example $(MAKEFLAGS)
+	$(MAKE) clean -C example
 
 .PHONY: all install \
   dist dist-mkdir dist-copy-files dist-package \
