diff -Naur b/projects/mozilla/control/npolibvlc.cpp a/projects/mozilla/control/npolibvlc.cpp
--- b/projects/mozilla/control/npolibvlc.cpp	2010-04-28 04:44:27.000000000 +0700
+++ a/projects/mozilla/control/npolibvlc.cpp	2010-07-26 13:22:10.357633997 +0700
@@ -52,7 +52,7 @@
     NPVariant res;
 
     if (NPVARIANT_IS_STRING(original))
-        STRINGZ_TO_NPVARIANT(strdup(NPVARIANT_TO_STRING(original).UTF8Characters), res);
+        STRINGZ_TO_NPVARIANT(strdup(NPVARIANT_TO_STRING(original).utf8characters), res);
     else if (NPVARIANT_IS_INT32(original))
         INT32_TO_NPVARIANT(NPVARIANT_TO_INT32(original), res);
     else if (NPVARIANT_IS_DOUBLE(original))
@@ -962,7 +962,7 @@
 void LibvlcPlaylistNPObject::parseOptions(const NPString &nps,
                                          int *i_options, char*** ppsz_options)
 {
-    if( nps.UTF8Length )
+    if( nps.utf8length )
     {
         char *s = stringValue(nps);
         char *val = s;
@@ -974,7 +974,7 @@
             {
                 int nOptions = 0;
 
-                char *end = val + nps.UTF8Length;
+                char *end = val + nps.utf8length;
                 while( val < end )
                 {
                     // skip leading blanks
@@ -1634,7 +1634,7 @@
 
     case ID_marquee_position:
         if( !NPVARIANT_IS_STRING(value) ||
-            !position_byname( NPVARIANT_TO_STRING(value).UTF8Characters, i ) )
+            !position_byname( NPVARIANT_TO_STRING(value).utf8characters, i ) )
             return INVOKERESULT_INVALID_VALUE;
 
         libvlc_video_set_marquee_int(p_md, libvlc_marquee_Position, i);
@@ -1780,7 +1780,7 @@
 
     case ID_logo_position:
         if( !NPVARIANT_IS_STRING(value) ||
-            !position_byname( NPVARIANT_TO_STRING(value).UTF8Characters, i ) )
+            !position_byname( NPVARIANT_TO_STRING(value).utf8characters, i ) )
             return INVOKERESULT_INVALID_VALUE;
 
         libvlc_video_set_logo_int(p_md, libvlc_logo_position, i);
@@ -1838,7 +1838,7 @@
         {
             if( !NPVARIANT_IS_STRING(args[i]) )
                 return INVOKERESULT_INVALID_VALUE;
-            len+=NPVARIANT_TO_STRING(args[i]).UTF8Length+1;
+            len+=NPVARIANT_TO_STRING(args[i]).utf8length+1;
         }
 
         buf = (char *)malloc( len+1 );
@@ -1848,8 +1848,8 @@
         for( h=buf,i=0;i<argCount;++i )
         {
             if(i) *h++=';';
-            len=NPVARIANT_TO_STRING(args[i]).UTF8Length;
-            memcpy(h,NPVARIANT_TO_STRING(args[i]).UTF8Characters,len);
+            len=NPVARIANT_TO_STRING(args[i]).utf8length;
+            memcpy(h,NPVARIANT_TO_STRING(args[i]).utf8characters,len);
             h+=len;
         }
         *h='\0';
diff -Naur b/projects/mozilla/control/nporuntime.cpp a/projects/mozilla/control/nporuntime.cpp
--- b/projects/mozilla/control/nporuntime.cpp	2010-04-28 04:44:27.000000000 +0700
+++ a/projects/mozilla/control/nporuntime.cpp	2010-07-26 13:22:20.654643889 +0700
@@ -32,11 +32,11 @@
 
 char* RuntimeNPObject::stringValue(const NPString &s)
 {
-    NPUTF8 *val = static_cast<NPUTF8*>(malloc((s.UTF8Length+1) * sizeof(*val)));
+    NPUTF8 *val = static_cast<NPUTF8*>(malloc((s.utf8length+1) * sizeof(*val)));
     if( val )
     {
-        strncpy(val, s.UTF8Characters, s.UTF8Length);
-        val[s.UTF8Length] = '\0';
+        strncpy(val, s.utf8characters, s.utf8length);
+        val[s.utf8length] = '\0';
     }
     return val;
 }
diff -Naur b/projects/mozilla/vlcplugin.cpp a/projects/mozilla/vlcplugin.cpp
--- b/projects/mozilla/vlcplugin.cpp	2010-06-08 05:57:45.000000000 +0700
+++ a/projects/mozilla/vlcplugin.cpp	2010-07-26 13:02:42.668635728 +0700
@@ -233,7 +233,7 @@
 
 bool EventObj::insert(const NPString &s, NPObject *l, bool b)
 {
-    event_t e = find_event(s.UTF8Characters);
+    event_t e = find_event(s.utf8characters);
     if( e>=maxbit() )
         return false;
 
@@ -258,7 +258,7 @@
 
 bool EventObj::remove(const NPString &s, NPObject *l, bool b)
 {
-    event_t e = find_event(s.UTF8Characters);
+    event_t e = find_event(s.utf8characters);
     if( e>=maxbit() || !get(e) )
         return false;
 
@@ -453,8 +453,8 @@
         NPString script;
         NPVariant result;
 
-        script.UTF8Characters = docLocHref;
-        script.UTF8Length = sizeof(docLocHref)-1;
+        script.utf8characters = docLocHref;
+        script.utf8length = sizeof(docLocHref)-1;
 
         if( NPN_Evaluate(p_browser, plugin, &script, &result) )
         {
@@ -462,11 +462,11 @@
             {
                 NPString &location = NPVARIANT_TO_STRING(result);
 
-                psz_baseURL = (char *) malloc(location.UTF8Length+1);
+                psz_baseURL = (char *) malloc(location.utf8length+1);
                 if( psz_baseURL )
                 {
-                    strncpy(psz_baseURL, location.UTF8Characters, location.UTF8Length);
-                    psz_baseURL[location.UTF8Length] = '\0';
+                    strncpy(psz_baseURL, location.utf8characters, location.utf8length);
+                    psz_baseURL[location.utf8length] = '\0';
                 }
             }
             NPN_ReleaseVariantValue(&result);
