Index: gir-repository-0.6.5/gir/Makefile.am
===================================================================
--- gir-repository-0.6.5.orig/gir/Makefile.am	2009-10-18 14:13:20.222891220 +0200
+++ gir-repository-0.6.5/gir/Makefile.am	2009-10-18 14:13:24.926892707 +0200
@@ -4,7 +4,7 @@ lib_LTLIBRARIES =
 BUILT_GIRSOURCES =
 CUSTOM_GIRSOURCES =
 XPATH_TESTS =
-COMMON_CUSTOM_LDFLAGS= -avoid-version -rdynamic
+COMMON_CUSTOM_LDFLAGS= -rdynamic
 
 AVAHI_GIRSOURCES = AvahiCore-0.6.gir
 if BUILD_AVAHI
Index: gir-repository-0.6.5/gir/Makefile.in
===================================================================
--- gir-repository-0.6.5.orig/gir/Makefile.in	2009-10-18 14:13:43.011399053 +0200
+++ gir-repository-0.6.5/gir/Makefile.in	2009-10-18 14:13:48.062896449 +0200
@@ -398,7 +398,7 @@ BUILT_GIRSOURCES = $(am__append_3) $(am_
 	$(am__append_39) $(am__append_40)
 CUSTOM_GIRSOURCES = $(am__append_1) $(am__append_2) $(am__append_17)
 XPATH_TESTS = $(am__append_14) $(am__append_18) $(am__append_24)
-COMMON_CUSTOM_LDFLAGS = -avoid-version -rdynamic
+COMMON_CUSTOM_LDFLAGS = -rdynamic
 AVAHI_GIRSOURCES = AvahiCore-0.6.gir
 MISSIONCONTROL_GIRSOURCES = DBusGLib-1.0.gir \
 			    TelepathyGLib.gir \
