--- a/gir/Makefile.am
+++ b/gir/Makefile.am
@@ -284,9 +284,8 @@
              --pkg atk \
              --pkg pango \
              --pkg gdk-$(GDK_TARGET)-2.0 \
-             -DGDK_COMPILATION \
 	     $(libgirepo_Gdk_custom_la_SOURCES) \
-             $(GDK_INCLUDEDIR)/gdk/*.h
+             $(GDK_INCLUDEDIR)/gdk/gdkx.h
 BUILT_GIRSOURCES += Gdk-2.0.gir
 
 lib_LTLIBRARIES += libgirepo-Gtk-custom.la
@@ -311,7 +310,6 @@
              --pkg gdk-$(GDK_TARGET)-2.0 \
              --pkg gtk+-$(GDK_TARGET)-2.0 \
              --output $@ \
-             -DGTK_COMPILATION \
              -D__GTK_FILE_CHOOSER_UTILS_H__ \
 	     -D__GTK_FILE_CHOOSER_PRIVATE_H__ \
              -D__GTK_FILE_SYSTEM_H__ \
@@ -321,7 +319,7 @@
              -D__GTK_TEXT_LAYOUT_H__ \
              -D__GTK_TEXT_DISPLAY_H__ \
              $(libgirepo_Gtk_custom_la_SOURCES) \
-             $(GTK_INCLUDEDIR)/gtk/*.h
+             $(GTK_INCLUDEDIR)/gtk/gtk.h
 BUILT_GIRSOURCES += Gtk-2.0.gir
 XPATH_TESTS += Gtk-2.0.xpath
 endif
