diff -Naur a/configure.in b/configure.in
--- a/configure.in	2011-07-01 23:40:13.000000000 +0700
+++ b/configure.in	2011-07-21 08:59:38.613180941 +0700
@@ -95,8 +95,13 @@
 fi
 
 if test "$GECKO_CFLAGS" = ""; then
-PKG_CHECK_MODULES(GECKO, [seamonkey-plugin seamonkey-xpcom],[ AC_MSG_NOTICE([Using seamonkey])
-    ],[AC_MSG_WARN([seamokey-plugin not found, trying another])])
+PKG_CHECK_MODULES(GECKO, [seamonkey-plugin < 1.9.2 seamonkey-xpcom < 1.9.2],[ AC_MSG_NOTICE([Using seamonkey < 1.9.2])
+    ],[AC_MSG_WARN([seamonkey-plugin < 1.9.2 not found, trying another])])
+fi
+
+if test "$GECKO_CFLAGS" = ""; then
+PKG_CHECK_MODULES(GECKO, [seamonkey-plugin >= 2 seamonkey-xpcom >= 2 ],[GECKO_CFLAGS="$GECKO_CFLAGS -DMOZ_NO_MOZALLOC" AC_DEFINE([HAVE_NEW_XULRUNNER],[1],[Define to 1 if using Seamonkey 2.0.0 or higher]) AC_MSG_NOTICE([Using seamonkey-plugin >= 2.0 ])
+    ],[AC_MSG_WARN([seamonkey-plugin not found, trying another])])
 fi
 
 if test "$GECKO_CFLAGS" = ""; then
