diff -ur mjpegtools-1.9.0.orig/configure.ac mjpegtools-1.9.0/configure.ac
--- mjpegtools-1.9.0.orig/configure.ac	2011-07-07 11:07:45.211089635 +0200
+++ mjpegtools-1.9.0/configure.ac	2011-07-07 11:18:42.513089556 +0200
@@ -155,11 +155,11 @@
 have_video4linux=false
 case $host in
   *-*-linux*)
-  AC_CHECK_HEADER(linux/videodev.h,
+  AC_CHECK_HEADER(libv4l1-videodev.h,
                   [have_video4linux=true
                    AC_DEFINE(HAVE_V4L, 1,
 		             [Building for Linux - using the video4linux API])],
-                  [AC_MSG_ERROR([videodev.h not found - please install the linux kernel headers])])
+                  [AC_MSG_ERROR([libv4l1-videodev.h not found - please install the v4l-utils devel package])])
    ;;
 esac
 AM_CONDITIONAL(HAVE_V4L, test x$have_video4linux = xtrue)
diff -ur mjpegtools-1.9.0.orig/lavtools/lavvideo.c mjpegtools-1.9.0/lavtools/lavvideo.c
--- mjpegtools-1.9.0.orig/lavtools/lavvideo.c	2011-07-07 11:07:45.220089635 +0200
+++ mjpegtools-1.9.0/lavtools/lavvideo.c	2011-07-07 11:18:27.601089556 +0200
@@ -63,7 +63,7 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include <videodev_mjpeg.h>
 #include <frequencies.h>
diff -ur mjpegtools-1.9.0.orig/lavtools/liblavplay.c mjpegtools-1.9.0/lavtools/liblavplay.c
--- mjpegtools-1.9.0.orig/lavtools/liblavplay.c	2011-07-07 11:07:45.221089635 +0200
+++ mjpegtools-1.9.0/lavtools/liblavplay.c	2011-07-07 11:16:01.227089575 +0200
@@ -68,7 +68,8 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE  192     /* 192-255 are private */
 #else
 #define VIDEO_MODE_PAL		0
 #define VIDEO_MODE_NTSC		1
diff -ur mjpegtools-1.9.0.orig/lavtools/liblavrec.c mjpegtools-1.9.0/lavtools/liblavrec.c
--- mjpegtools-1.9.0.orig/lavtools/liblavrec.c	2011-07-07 11:07:45.221089635 +0200
+++ mjpegtools-1.9.0/lavtools/liblavrec.c	2011-07-07 11:16:18.362089573 +0200
@@ -63,7 +63,8 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE  192     /* 192-255 are private */
 #ifdef HAVE_SYS_SOUNDCARD_H
 #include <sys/soundcard.h>
 #endif
diff -ur mjpegtools-1.9.0.orig/lavtools/testrec.c mjpegtools-1.9.0/lavtools/testrec.c
--- mjpegtools-1.9.0.orig/lavtools/testrec.c	2011-07-07 11:07:45.220089635 +0200
+++ mjpegtools-1.9.0/lavtools/testrec.c	2011-07-07 11:19:56.128089546 +0200
@@ -89,7 +89,7 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <linux/soundcard.h>
 
 /* These are explicit prototypes for the compiler, to prepare separation of audiolib.c */
