https://code.google.com/p/clementine-player/issues/detail?id=3847#c27

--- Clementine-1.2.3/src/library/libraryquery.cpp
+++ Clementine-1.2.3/src/library/libraryquery.cpp
@@ -119,7 +119,7 @@
   // which gives very poor performance. See
   // https://github.com/clementine-player/Clementine/pull/4285 for
   // more details.
-  where_clauses_ << QString("+effective_compilation = %1")
+  where_clauses_ << QString("+effective_compilation LIKE %1")
                         .arg(compilation ? 1 : 0);
 }
 
