--- clementine-1.1.1/src/main.cpp.orig	2013-06-14 17:33:48.686833768 +0700
+++ clementine-1.1.1/src/main.cpp	2013-06-14 17:34:37.130832514 +0700
@@ -265,7 +265,9 @@
 #endif
 
   // This makes us show up nicely in gnome-volume-control
-  g_type_init();
+#if !GLIB_CHECK_VERSION(2, 36, 0)
+  g_type_init();  // Deprecated in glib 2.36.0
+#endif
   g_set_application_name(QCoreApplication::applicationName().toLocal8Bit());
 
   RegisterMetaTypes();
