diff -urN guichan-0.8.2/include/guichan/key.hpp guichan-0.8.2.uh/include/guichan/key.hpp
--- guichan-0.8.2/include/guichan/key.hpp   2009-09-03 21:35:02.000000000 +0100
+++ guichan-0.8.2.uh/include/guichan/key.hpp    2010-01-11 23:34:52.000000000 +0000
@@ -125,7 +125,7 @@
             SPACE              = ' ',
             TAB                = '\t',
             ENTER              = '\n',
-            LEFT_ALT           = 1000,
+            LEFT_ALT           = -1000,
             RIGHT_ALT,
             LEFT_SHIFT,
             RIGHT_SHIFT,
