--- configure.in
+++ configure.in
@@ -77,6 +77,8 @@
 {
   OPENGL="yes"
   GUICHAN_EXTRADIRS="$GUICHAN_EXTRADIRS opengl"
+  GL_LIBS="`pkg-config --libs gl`"
+  AC_SUBST(GL_LIBS)
 }
 
 ForceOpenGL()
@@ -114,8 +116,9 @@
 HaveSDLImage()
 {
   SDLIMAGE="yes"
-  SDL_LIBS="$SDL_LIBS -lSDL_image"
+  SDL_LIBS="-lSDL_image `pkg-config --libs sdl`"
   CPPFLAGS="$CPPFLAGS `sdl-config --cflags`"
+  AC_SUBST(SDL_LIBS)
 }
 
 ForceSDLImage()
@@ -144,6 +147,8 @@
 {
   ALLEGRO="yes"
   GUICHAN_EXTRADIRS="$GUICHAN_EXTRADIRS allegro"
+  ALLEGRO_LIBS="`allegro-config --libs`"
+  AC_SUBST(ALLEGRO_LIBS)
 }
 
 ForceAllegro()
--- src/allegro/Makefile.am
+++ src/allegro/Makefile.am
@@ -2,7 +2,9 @@
 
 INCLUDES = -I$(top_srcdir)/include
 
-libguichan_allegro_la_LDFLAGS = -no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libguichan_allegro_la_LIBADD = $(ALLEGRO_LIBS) ../libguichan.la
+
+libguichan_allegro_la_LDFLAGS = -Wl,--no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 libguichan_allegro_la_SOURCES =        \
        allegro.cpp     \
--- src/Makefile.am
+++ src/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = widgets @GUICHAN_EXTRADIRS@ hge openlayer contrib
+SUBDIRS = widgets . @GUICHAN_EXTRADIRS@ hge openlayer contrib
 
 INCLUDES = -I$(top_srcdir)/include
 
@@ -6,7 +6,7 @@
 
 libguichan_la_LIBADD = widgets/libguichan_widgets.la
 
-libguichan_la_LDFLAGS = -no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 
+libguichan_la_LDFLAGS = -Wl,--no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 
 
 libguichan_la_DEPENDENCIES = \
    widgets/libguichan_widgets.la
--- src/opengl/Makefile.am
+++ src/opengl/Makefile.am
@@ -2,7 +2,9 @@
 
 INCLUDES = -I$(top_srcdir)/include
 
-libguichan_opengl_la_LDFLAGS = -no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 
+libguichan_opengl_la_LIBADD = $(GL_LIBS) ../libguichan.la
+
+libguichan_opengl_la_LDFLAGS = -Wl,--no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 
 
 libguichan_opengl_la_SOURCES = \
    opengl.cpp      \
--- src/sdl/Makefile.am
+++ src/sdl/Makefile.am
@@ -4,7 +4,9 @@
 
 INCLUDES = -I$(top_srcdir)/include
 
-libguichan_sdl_la_LDFLAGS = -no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libguichan_sdl_la_LIBADD = $(SDL_LIBS) ../libguichan.la
+
+libguichan_sdl_la_LDFLAGS = -Wl,--no-undefined -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 libguichan_sdl_la_SOURCES =    \
    sdl.cpp         \
