--- speech_tools/base_class/EST_TSimpleMatrix.cc.orig	2012-07-12 14:14:02.322259519 +0700
+++ speech_tools/base_class/EST_TSimpleMatrix.cc	2012-07-12 14:54:49.086196153 +0700
@@ -45,6 +45,7 @@
 #include <fstream>
 #include <iostream>
 #include "EST_cutils.h"
+#include <string.h>
 
 template<class T> 
 void EST_TSimpleMatrix<T>::copy_data(const EST_TSimpleMatrix<T> &a)
--- speech_tools/base_class/EST_TSimpleVector.cc.orig	2012-07-12 14:20:26.779249562 +0700
+++ speech_tools/base_class/EST_TSimpleVector.cc	2012-07-12 14:57:56.432191300 +0700
@@ -44,6 +44,7 @@
 #include "EST_matrix_support.h"
 #include <fstream>
 #include "EST_cutils.h"
+#include <string.h>
 
 template<class T> void EST_TSimpleVector<T>::copy(const EST_TSimpleVector<T> &a)
 {
