--- kipi-plugins-1.9.0/printimages/wizard.cpp.orig	2012-07-10 12:43:46.692335446 +0700
+++ kipi-plugins-1.9.0/printimages/wizard.cpp	2012-07-10 12:42:56.444336764 +0700
@@ -88,7 +88,7 @@
     WizardPage ( KAssistantDialog* dialog, const QString& title )
         : QWidget ( dialog )
     {
-        setupUi ( this );
+        this->setupUi ( this );
         layout()->setMargin ( 0 );
         mPage = dialog->addPage ( this, title );
     }
--- kipi-plugins-1.9.0/htmlexport/wizard.cpp.orig	2012-07-10 13:03:36.029304734 +0700
+++ kipi-plugins-1.9.0/htmlexport/wizard.cpp	2012-07-10 13:03:55.759304229 +0700
@@ -78,7 +78,7 @@
 public:
     WizardPage(KAssistantDialog* dialog, const QString& title)
     : QWidget(dialog) {
-        setupUi(this);
+        this->setupUi(this);
         layout()->setMargin(0);
         mPage = dialog->addPage(this, title);
     }
