From 8e67a5f03e159911a3c5cde604871fea1f2b5f91 Mon Sep 17 00:00:00 2001
From: Javier Llorente <javier@opensuse.org>
Date: Thu, 19 Mar 2015 21:46:47 +0100
Subject: [PATCH 03/36] fix bug #1625

---
 src/lib/app/mainmenu.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/app/mainmenu.cpp b/src/lib/app/mainmenu.cpp
index 9f2bec0..2b5e856 100644
--- a/src/lib/app/mainmenu.cpp
+++ b/src/lib/app/mainmenu.cpp
@@ -38,6 +38,7 @@
 #include <QStatusBar>
 #include <QWebPage>
 #include <QMenuBar>
+#include <QDesktopServices>
 
 #ifdef Q_OS_MAC
 extern void qt_mac_set_dock_menu(QMenu* menu);
@@ -175,9 +176,8 @@ void MainMenu::savePageScreen()
 
 void MainMenu::sendLink()
 {
-    if (m_window) {
-        m_window->weView()->savePageAs();
-    }
+    const QUrl mailUrl = QUrl::fromEncoded("mailto:%20?body=" + QUrl::toPercentEncoding(m_window->weView()->url().toEncoded()) + "&subject=" + QUrl::toPercentEncoding(m_window->weView()->title()));
+    QDesktopServices::openUrl(mailUrl);
 }
 
 void MainMenu::printPage()
-- 
2.5.1

