From 693be09ae73673298c800eed5127e19c61303b7e Mon Sep 17 00:00:00 2001
From: Javier Llorente <javier@opensuse.org>
Date: Fri, 17 Apr 2015 16:24:20 +0200
Subject: [PATCH 09/36] change wording from 'on this site' to 'for this site'

---
 src/lib/network/networkmanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/network/networkmanager.cpp b/src/lib/network/networkmanager.cpp
index 6497289..a0f55df 100644
--- a/src/lib/network/networkmanager.cpp
+++ b/src/lib/network/networkmanager.cpp
@@ -304,7 +304,7 @@ void NetworkManager::authentication(QNetworkReply* reply, QAuthenticator* auth)
     QLineEdit* user = new QLineEdit(dialog);
     QLineEdit* pass = new QLineEdit(dialog);
     QCheckBox* save = new QCheckBox(dialog);
-    save->setText(tr("Save username and password on this site"));
+    save->setText(tr("Save username and password for this site"));
     pass->setEchoMode(QLineEdit::Password);
 
     QDialogButtonBox* box = new QDialogButtonBox(dialog);
-- 
2.5.1

