From 32f5c007795627445b90a23c66bbcb86678e3705 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Thu, 21 May 2015 17:17:16 +0200
Subject: [PATCH 11/36] KWalletPasswords: Fix build with KF5 KWallet

Closes #1545
---
 src/plugins/KWalletPasswords/kwalletpasswordbackend.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/plugins/KWalletPasswords/kwalletpasswordbackend.h b/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
index db29afe..21c0f85 100644
--- a/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
+++ b/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
@@ -19,7 +19,12 @@
 #define KWALLETPASSWORDBACKEND_H
 
 #include <QVector>
+
+#if QT_VERSION >= 0x050000
+#include <KF5/KWallet/Wallet>
+#else
 #include <KDE/KWallet/Wallet>
+#endif
 
 #include "passwordbackends/passwordbackend.h"
 #include "passwordmanager.h"
-- 
2.5.1

