From bdcd3269cb86bea0f23c5a566604f843ac9cbff6 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Thu, 21 May 2015 17:26:18 +0200
Subject: [PATCH 12/36] Fix include for KF5 KWallet

See #1545
---
 src/plugins/KWalletPasswords/kwalletpasswordbackend.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/KWalletPasswords/kwalletpasswordbackend.h b/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
index 21c0f85..23623f5 100644
--- a/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
+++ b/src/plugins/KWalletPasswords/kwalletpasswordbackend.h
@@ -21,7 +21,7 @@
 #include <QVector>
 
 #if QT_VERSION >= 0x050000
-#include <KF5/KWallet/Wallet>
+#include <KF5/KWallet/KWallet>
 #else
 #include <KDE/KWallet/Wallet>
 #endif
-- 
2.5.1

