From c4c31ac5a103bed3f29d24dc6cceb49970f1503b Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Thu, 11 Jun 2015 21:07:11 +0200
Subject: [PATCH 23/36] TabBar: Don't use ProxyStyle directly

---
 src/lib/tabwidget/tabbar.cpp | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/lib/tabwidget/tabbar.cpp b/src/lib/tabwidget/tabbar.cpp
index 0079014..990a55e 100644
--- a/src/lib/tabwidget/tabbar.cpp
+++ b/src/lib/tabwidget/tabbar.cpp
@@ -25,7 +25,6 @@
 #include "tabbedwebview.h"
 #include "mainapplication.h"
 #include "pluginproxy.h"
-#include "proxystyle.h"
 #include "iconprovider.h"
 
 #include <QMenu>
@@ -150,11 +149,9 @@ void TabBar::closeAllButCurrent()
 
 QSize TabBar::tabSizeHint(int index, bool fast) const
 {
-    if (!isVisible() || !mApp->proxyStyle()) {
+    if (!isVisible()) {
         // Don't calculate it when tabbar is not visible
         // It produces invalid size anyway
-        //
-        // We also need ProxyStyle to be set before calculating minimum sizes for tabs
         return QSize(-1, -1);
     }
 
@@ -266,13 +263,12 @@ QSize TabBar::tabSizeHint(int index, bool fast) const
 
 int TabBar::comboTabBarPixelMetric(ComboTabBar::SizeType sizeType) const
 {
-    if (!mApp->proxyStyle() || !isVisible()) {
+    if (!isVisible())
         return -1;
-    }
 
     switch (sizeType) {
     case ComboTabBar::PinnedTabWidth:
-        return 16 + mApp->proxyStyle()->pixelMetric(QStyle::PM_TabBarTabHSpace, 0, this);
+        return 16 + style()->pixelMetric(QStyle::PM_TabBarTabHSpace, 0, this);
 
     case ComboTabBar::ActiveTabMinimumWidth:
     case ComboTabBar::NormalTabMinimumWidth:
-- 
2.5.1

