--- zvbi-0.2.33/src/io-dvb.c.orig	2008-02-24 15:18:21.000000000 +0100
+++ zvbi-0.2.33/src/io-dvb.c	2011-11-06 20:54:41.805070761 +0100
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <sys/select.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 
 #ifndef HAVE_S64_U64
   /* Linux 2.6.x asm/types.h defines __s64 and __u64 only
--- zvbi-0.2.33/src/io-v4l.c.orig	2008-02-24 15:17:28.000000000 +0100
+++ zvbi-0.2.33/src/io-v4l.c	2011-11-06 20:54:23.938403497 +0100
@@ -43,6 +43,7 @@
 #include <sys/time.h>		/* timeval */
 #include <sys/types.h>		/* fd_set, uid_t */
 #include <sys/ioctl.h>		/* for (_)videodev.h */
+#include <sys/stat.h>
 #include <pthread.h>
 
 #include "videodev.h"
--- zvbi-0.2.33/contrib/ntsc-cc.c.orig	2007-11-27 18:39:34.000000000 +0100
+++ zvbi-0.2.33/contrib/ntsc-cc.c	2011-11-06 20:58:27.808411663 +0100
@@ -35,6 +35,7 @@
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/time.h>
+#include <sys/stat.h>
 #ifdef HAVE_GETOPT_LONG
 #  include <getopt.h>
 #endif
