--- src/nvtt/CompressRGB.cpp.orig	2013-03-07 03:17:17.369126322 +0700
+++ src/nvtt/CompressRGB.cpp	2013-03-07 03:18:04.989125088 +0700
@@ -90,7 +90,7 @@
 	// Determine pitch.
 	uint pitch = computePitch(w, compressionOptions.bitcount);
 
-	uint8 * dst = (uint8 *)mem::malloc(pitch + 4);
+	uint8 * dst = (uint8 *)::malloc(pitch + 4);
 
 	for (uint y = 0; y < h; y++)
 	{
@@ -135,6 +135,6 @@
 		}
 	}
 
-	mem::free(dst);
+	::free(dst);
 }
 
