Index: examples/wxPlayer/MIDIDeviceDialog.h
===================================================================
--- examples/wxPlayer/MIDIDeviceDialog.h	(revisión: 683)
+++ examples/wxPlayer/MIDIDeviceDialog.h	(copia de trabajo)
@@ -15,7 +15,7 @@
   }
 
 private:
-  void MIDIDeviceDialog::OnButton(wxCommandEvent& event);
+  void OnButton(wxCommandEvent& event);
 
   wxTextCtrl* m_name;
 
Index: examples/wxPlayer/DeviceFrame.cpp
===================================================================
--- examples/wxPlayer/DeviceFrame.cpp	(revisión: 683)
+++ examples/wxPlayer/DeviceFrame.cpp	(copia de trabajo)
@@ -57,7 +57,7 @@
 
 
 DeviceFrame::DeviceFrame(audiere::AudioDevicePtr device)
-: wxMDIParentFrame(0, -1, wxT("Audio Device - " + wxString(device->getName())),
+: wxMDIParentFrame(0, -1, wxT("Audio Device - " + CStr2wxString( device->getName() )),
                    wxDefaultPosition, wxSize(400, 500))
 {
   m_device = device;
diff -urN audiere-1.9.4/examples/wxPlayer/TagsDialog.h audiere-1.9.4-chenz/examples/wxPlayer/TagsDialog.h
--- examples/wxPlayer/TagsDialog.h	1970-01-01 01:00:00.000000000 +0100
+++ examples/wxPlayer/TagsDialog.h	2007-03-27 10:56:50.000000000 +0200
@@ -0,0 +1,26 @@
+#ifndef TAGS_DIALOG_H
+#define TAGS_DIALOG_H
+
+
+#include <audiere.h>
+#include "wx.h"
+
+
+class TagsDialog : public wxDialog {
+public:
+  TagsDialog(
+    wxWindow* parent,
+    audiere::SampleSourcePtr source);
+
+private:
+  void OnButton(wxCommandEvent& event);
+  void OnClose(wxCloseEvent& event);
+
+  wxListBox* m_tags;
+  wxButton* m_close;
+
+  DECLARE_EVENT_TABLE()
+};
+
+
+#endif
