--- OpenCV-2.1.0/src/highgui/cvcap_ffmpeg.cpp.orig	2010-10-19 12:23:43.894000000 +0700
+++ OpenCV-2.1.0/src/highgui/cvcap_ffmpeg.cpp	2010-10-19 12:56:16.828999776 +0700
@@ -48,8 +48,9 @@
 extern "C" {
 #if !defined(WIN32) || defined(__MINGW32__)
 // some versions of FFMPEG assume a C99 compiler, and don't define INT64_C
-#ifndef INT64_C
+#if !defined INT64_C || !defined UINT64_C
 #define INT64_C
+#define UINT64_C
 #define __STDC_CONSTANT_MACROS
 // force re-inclusion of stdint.h to get INT64_C macro
 #undef _STDINT_H
