diff -uNr celestia-1.6.1/src/celengine/frametree.h celestia-1.6.1p/src/celengine/frametree.h
--- celestia-1.6.1/src/celengine/frametree.h	
+++ celestia-1.6.1p/src/celengine/frametree.h	
@@ -13,6 +13,7 @@
 #ifndef _CELENGINE_FRAMETREE_H_
 #define _CELENGINE_FRAMETREE_H_
 
+#include <unistd.h>
 #include <vector>
 #include <cstdio>
 
diff -uNr celestia-1.6.1/src/celengine/planetgrid.cpp celestia-1.6.1p/src/celengine/planetgrid.cpp
--- celestia-1.6.1/src/celengine/planetgrid.cpp	
+++ celestia-1.6.1p/src/celengine/planetgrid.cpp	
@@ -12,6 +12,7 @@
 
 #include <cstdio>
 #include <cmath>
+#include <celmath/mathlib.h>
 #include <celmath/intersect.h>
 #include "planetgrid.h"
 #include "body.h"
