Index: Source/WebCore/plugins/gtk/PluginViewGtk.cpp
===================================================================
--- Source/WebCore/plugins/gtk/PluginViewGtk.cpp	(revision 97584)
+++ Source/WebCore/plugins/gtk/PluginViewGtk.cpp	(working copy)
@@ -70,6 +70,7 @@
 #endif
 #include <gtk/gtk.h>

 #if defined(XP_UNIX)
+#define String XtStringType
 #include "RefPtrCairo.h"
 #include "gtk2xtbin.h"
@@ -82,6 +83,7 @@
 #include "PluginMessageThrottlerWin.h"
 #include <gdk/gdkwin32.h>
+#undef String
 #endif

 using JSC::ExecState;
 using JSC::Interpreter;
