diff -Nur webkit-1.6.1.orig/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp webkit-1.6.1/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp
--- webkit-1.6.1.orig/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp	2011-09-26 14:55:03.000000000 -0600
+++ webkit-1.6.1/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp	2012-01-20 23:06:40.282863869 -0700
@@ -28,6 +28,7 @@
 #include "PluginObject.h"
 #include <assert.h>
 #include <string.h>
+#include <unistd.h>
 
 using namespace std;
 extern NPNetscapeFuncs *browser;
