--- a/src/gallium/state_trackers/clover/llvm/invocation.cpp	2015-04-05 14:06:56.890565202 +0700
+++ b/src/gallium/state_trackers/clover/llvm/invocation.cpp	2015-04-05 19:06:16.006663973 +0700
@@ -79,6 +79,8 @@
 #include <sstream>
 #include <libelf.h>
 #include <gelf.h>
+#include <string>
+#include <utility>

 using namespace clover;

@@ -128,7 +130,7 @@
       if (!strcmp("stderr", dbg_file)) {
          std::cerr << msg;
        } else {
-        std::ofstream file(dbg_file + suffix, std::ios::app);
+        std::ofstream file(dbg_file, std::ios::app);
         file << msg;
        }
    }
