--- libtirpc-0.2.1/configure.ac.orig	2009-11-30 15:04:55.000000000 +0100
+++ libtirpc-0.2.1/configure.ac	2010-08-14 08:49:50.618819943 +0200
@@ -3,16 +3,23 @@
 AM_MAINTAINER_MODE
 AC_CONFIG_SRCDIR([src/auth_des.c])
 
+GSSAPI_IMPLEMENTATION=libgssglue
 AC_ARG_ENABLE(gss,[  --enable-gss            Turn on gss api], [case "${enableval}" in
-        yes) gss=true ; AC_CHECK_LIB([gssapi],[gss_init_sec_context]) ;;
+        yes) gss=true ;;
         no)  gss=false ;;
-        *) AC_MSG_ERROR(bad value ${enableval} for --enable-gss) ;;
+        *) gss=true; GSSAPI_IMPLEMENTATION="$enableval" ;;
       esac],[gss=false])
 AM_CONDITIONAL(GSS, test x$gss = xtrue)
 if test x$gss = xtrue; then
-	AC_DEFINE(HAVE_LIBGSSAPI, 1, [])
-	PKG_CHECK_MODULES(GSSGLUE, libgssglue, [],
-	AC_MSG_ERROR([Unable to locate information required to use libgssglue.]))
+	AC_DEFINE(HAVE_LIBGSSAPI, 1, [Have GSSAPI])
+	PKG_CHECK_MODULES([GSSGLUE], [$GSSAPI_IMPLEMENTATION],
+		[echo GSSAPI in $GSSAPI_IMPLEMENTATION],
+		[PKG_CHECK_MODULES([GSSGLUE], [libgssglue],
+			[echo GSSGLUE found; GSSAPI_IMPLEMENTATION=libgssglue],
+			[PKG_CHECK_MODULES([GSSGLUE], [heimdal-gssapi],
+				[echo HEIMDAL found; GSSAPI_IMPLEMENTATION=heimdal-gssapi],
+				[AC_MSG_ERROR([Unable to locate information required to use libgssglue.])])])])
+AC_SUBST([GSSAPI_IMPLEMENTATION])
 fi
 
 AC_PROG_CC
diff -ur libtirpc-0.1.10/libtirpc.pc.in libtirpc-0.1.10-heimdal/libtirpc.pc.in
--- libtirpc-0.1.10/libtirpc.pc.in	2008-11-19 14:01:43.000000000 +0100
+++ libtirpc-0.1.10-heimdal/libtirpc.pc.in	2009-06-30 14:55:59.811577076 +0200
@@ -5,7 +5,7 @@
 
 Name: libtirpc
 Description: Transport Independent RPC Library
-Requires:
+Requires: @GSSAPI_IMPLEMENTATION@
 Version: @PACKAGE_VERSION@
 Libs: -L@libdir@ -ltirpc
 Cflags: -I@includedir@/tirpc
