--- libtirpc-0.2.2/src/Makefile.am.orig	2011-06-12 17:44:39.795059643 +0200
+++ libtirpc-0.2.2/src/Makefile.am	2011-06-12 19:50:08.598645068 +0200
@@ -51,7 +51,10 @@
         rpc_callmsg.c rpc_generic.c rpc_soc.c rpcb_clnt.c rpcb_prot.c \
         rpcb_st_xdr.c svc.c svc_auth.c svc_dg.c svc_auth_unix.c svc_generic.c \
         svc_raw.c svc_run.c svc_simple.c svc_vc.c getpeereid.c \
-        auth_time.c auth_des.c authdes_prot.c des_crypt.c
+        auth_time.c auth_des.c authdes_prot.c
+
+## formerly in glibc
+libtirpc_la_SOURCES += des_crypt.c crypt_client.c key_call.c netname.c rpcdname.c
 
 ## XDR
 libtirpc_la_SOURCES += xdr.c xdr_rec.c xdr_array.c xdr_float.c xdr_mem.c xdr_reference.c xdr_stdio.c
