diff -ur libtirpc-0.2.0/src/auth_gss.c libtirpc-0.2.0-test/src/auth_gss.c
--- libtirpc-0.2.0/src/auth_gss.c	2009-05-28 21:25:26.000000000 +0200
+++ libtirpc-0.2.0-test/src/auth_gss.c	2009-07-01 21:01:32.818065056 +0200
@@ -309,6 +309,10 @@
 	}
 	/* Checksum serialized RPC header, up to and including credential. */
 	rpcbuf.length = XDR_GETPOS(xdrs);
+	/* XDR_GETPOS most of the time returns -1 (you can't seek on a socket),
+	 * and gss_get_mic doesn't like it */
+	if ((int)rpcbuf.length == -1)
+		rpcbuf.length = 0;
 	XDR_SETPOS(xdrs, 0);
 	rpcbuf.value = XDR_INLINE(xdrs, rpcbuf.length);
 
