--- a/configure.in	2011-03-10 16:54:19.000000000 +0700
+++ b/configure.in	2011-11-28 09:16:36.478000135 +0700
@@ -554,7 +554,7 @@
 
 AC_CHECK_FUNC(xdrmem_create, XDRLIB=, [
 AC_CHECK_LIB(sun, xdrmem_create, XDRLIB=-lsun, [
-AC_CHECK_LIB(nsl, xdrmem_create, XDRLIB=-lnsl, [
+AC_CHECK_LIB(tirpc, xdrmem_create, XDRLIB=-ltirpc, [
 AC_CHECK_LIB(rpclib, xdrmem_create, XDRLIB=-lrpclib, [
 AC_CHECK_LIB(xdr, xdrmem_create, XDRLIB="-lxdr -liberty -lws2_32" , 
    [ AC_MSG_ERROR([*** Unable to locate XDR functions.]) ], -liberty -lws2_32 )
