--- grass-6.4.1/configure.in.orig	2011-11-27 11:29:08.783000719 +0700
+++ grass-6.4.1/configure.in	2011-11-27 11:04:39.886000542 +0700
@@ -552,14 +552,14 @@
 
 # Note: -lxdr -liberty -lws2_32 is for MinGW
 
-AC_CHECK_FUNC(xdrmem_create, XDRLIB=, [
-AC_CHECK_LIB(sun, xdrmem_create, XDRLIB=-lsun, [
-AC_CHECK_LIB(nsl, xdrmem_create, XDRLIB=-lnsl, [
-AC_CHECK_LIB(rpclib, xdrmem_create, XDRLIB=-lrpclib, [
-AC_CHECK_LIB(xdr, xdrmem_create, XDRLIB="-lxdr -liberty -lws2_32" , 
-   [ AC_MSG_ERROR([*** Unable to locate XDR functions.]) ], -liberty -lws2_32 )
-])])])])
-AC_SUBST(XDRLIB)
+#AC_CHECK_FUNC(xdrmem_create, XDRLIB=, [
+#AC_CHECK_LIB(sun, xdrmem_create, XDRLIB=-lsun, [
+#AC_CHECK_LIB(nsl, xdrmem_create, XDRLIB=-lnsl, [
+#AC_CHECK_LIB(rpclib, xdrmem_create, XDRLIB=-lrpclib, [
+#AC_CHECK_LIB(xdr, xdrmem_create, XDRLIB="-lxdr -liberty -lws2_32" , 
+#   [ AC_MSG_ERROR([*** Unable to locate XDR functions.]) ], -liberty -lws2_32 )
+#])])])])
+#AC_SUBST(XDRLIB)
 
 AC_CHECK_FUNCS(asprintf)
 AC_SUBST(HAVE_ASPRINTF)
